### DISCUSSION TITLE
basedpyright highlighting

### DISCUSSION THREAD

**Discussion by Dronakurl** [LazyVim/LazyVim#3848] [Q&A]
I just installed the "python" lazyvim extra and used basedpyright just as it is described here
https://www.lazyvim.org/extras/lang/python

I noticed, that some highlights are different, e.g. the "self" variable in class definitions is not recognized as a builtin.
Is there a way to configure basedpyright, so "self" is identified as a builtin?

I made sure to use the latest version (using bob, 0.10.0) and used a fresh install of LazyVim in an extra folder (using NVIM_APPNAME to start it).

With basedpyright
![image](https://github.com/LazyVim/LazyVim/assets/76913264/3f25ca64-3a98-4177-bec2-412c8c862160)

With pyright
![image](https://github.com/LazyVim/LazyVim/assets/76913264/966e7fcc-3f6e-4df4-b0c9-09d90b20508f)

**Comment by xzbdmw** [comment]
I guess it is overwritten by lsp semantic highlight, you can put cursor in self and run :Inspect
